src/test/java/org/kiwiproject/jersey/client/RegistryAwareClientTest.java

```diff
@@ -0,0 +1,124 @@
+package org.kiwiproject.jersey.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.kiwiproject.registry.client.RegistryClient;
+import org.kiwiproject.registry.model.Port;
+import org.kiwiproject.registry.model.ServiceInstance;
+import org.kiwiproject.registry.model.ServicePaths;
+
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+import java.util.List;
+import java.util.Optional;
+
+@DisplayName("RegistryAwareClient")
+class RegistryAwareClientTest {
```

Comment:
Add tests for validation and default values

Reply:
This was actually for ServiceIdentifier, but I added a whole set of tests